e2e/test/scenarios/joins/reproductions/22859-multi-nested-joins-wrong-aliasing.cy.spec.js

```diff
@@ -31,44 +31,53 @@ const questionDetails = {
 
 describe("issue 22859 - multiple levels of nesting", () => {
   beforeEach(() => {
-    cy.intercept("POST", "/api/card").as("saveQuestion");
-
     restore();
     cy.signInAsAdmin();
 
     cy.createQuestion(questionDetails, { wrapId: true, idAlias: "q1Id" });
 
     // Join Orders table with the previously saved question and save it again
-    openOrdersTable({ mode: "notebook" });
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText("Join data").click();
-
-    popover().within(() => {
-      cy.findByText("Sample Database").click();
-      cy.findByText("Saved Questions").click();
-      cy.findByText(questionDetails.name).click();
+    cy.get("@q1Id").then(id => {
+      const nestedQuestionDetails = {
+        name: "22859-Q2",
+        query: {
+          "source-table": ORDERS_ID,
+          joins: [
+            {
+              fields: "all",
+              alias: `Question ${id}`,
+              condition: [
+                "=",
+                ["field", ORDERS.PRODUCT_ID, { "base-type": "type/Integer" }],
+                [
+                  "field",
+                  REVIEWS.PRODUCT_ID,
+                  {
+                    "base-type": "type/Integer",
+                    "join-alias": `Question ${id}`,
+                  },
+                ],
+              ],
+              "source-table": `card__${id}`,
+            },
+          ],
+          limit: 5,
+        },
+      };
+
+      cy.createQuestion(nestedQuestionDetails, {
+        wrapId: true,
+        idAlias: "q2Id",
+      });
     });
-
-    popover().contains("Product ID").click();
-
-    popover().contains("Product ID").click();
-
-    visualize();
-
-    saveQuestion("22859-Q2");
-
-    cy.wait("@saveQuestion").then(({ response: { body } }) =>
-      cy.wrap(body.id).as("q2Id"),
-    );
-
-    getJoinedTableColumnHeader();
   });
 
   it("model based on multi-level nested saved question should work (metabase#22859-1)", () => {
     cy.get("@q2Id").then(id => {
       // Convert the second question to a model
       cy.request("PUT", `/api/card/${id}`, { dataset: true });
 
+      cy.intercept("POST", "/api/dataset").as("dataset");
       cy.visit(`/model/${id}`);
       cy.wait("@dataset");
```

Comment:
hmm, how could we wait for dataset if it wasn't intercepted? I believe `cypress` would fail if you didn't intercept.

Reply:
`openOrdersTable` helper (that's not removed) had this intercept internally.